### PR TITLE
Introduce Input enum for symbolization APIs

### PR DESCRIPTION
**Pull request by danielocfb**

### CODE REVIEW

**Comment by danielocfb** [comment]
This is just a sneak peek at this point. Naming suggestions would be welcome.

File: src/symbolize/mod.rs [hunk 1]
```diff
@@ -117,6 +117,29 @@ pub use symbolizer::Symbolizer;
...
+    /// A file address.
+    ///
+    /// A file address is an address as it would appear in a binary or
+    /// debug symbol file, i.e., one excluding any relocations.
+    FileAddr(A),
+    /// A file offset.
+    ///
+    /// A file offset is the linear offset of a byte in a file in the file
+    /// system from its start.
+    FileOffset(O),
```
Comment (anakryiko) [inline src/symbolize/mod.rs:140]: tbh, I find FileAddr confusing. My suggestion would be:

VirtAddr -> AbsAddr (absolute address) or perhaps just Addr
FileAddr -> VirtOffset (perhaps could be VirtAddr to match ELF terminology, I guess, though with shared libraries and PIE executables "address" is a lie and it is actually offset within virtual address space).
FileOffset -> FileOffset

From user space perspective there is no "virtual" address, it's all real/absolute addresses for them. It's only kernel that knows about physical and virtual addresses. For me it's more about offsets vs absolute addresses, and then we get two kinds of offsets, one file contents and another is, for lack of better term, virtual offset.

File: src/symbolize/mod.rs [hunk 2]
```diff
@@ -117,6 +118,29 @@ pub use symbolizer::Symbolizer;
...
+pub enum Input<A, O> {
+    /// A virtual address.
+    ///
+    /// A virtual address is an address as a process would see it, for
+    /// example. A virtual address may contain relocation or address space
+    /// randomization effects.
+    VirtAddr(A),
+    /// A file address.
+    ///
+    /// A file address is an address as it would appear in a binary or
```
Comment (anakryiko) [inline src/symbolize/mod.rs:133]: it's not just about relocations, so it feels a bit too much to go into specific (but incomplete) details. Maybe we can just say that this is address (or offset if you agree with me below) that matches ELF file (and DWARF) symbol and segment information.

File: src/symbolize/mod.rs [hunk 3]
```diff
@@ -117,6 +118,29 @@ pub use symbolizer::Symbolizer;
...
+    /// debug symbol file, i.e., one excluding any relocations.
+    FileAddr(A),
+    /// A file offset.
+    ///
+    /// A file offset is the linear offset of a byte in a file in the file
```
Comment (anakryiko) [inline src/symbolize/mod.rs:138]: "in a file in the file system from its start" reads a bit awkwardly... "A file offset within the corresponding ELF binary"?

**Review by anakryiko — COMMENTED** [review]
naming is hard, but left my thoughts

**Comment by danielocfb** [comment]
Will open a new PR.

**Pull request closed without merging**